appender.go

```diff
@@ -85,6 +87,8 @@ type Appender struct {
 	metrics               metrics
 	mu                    sync.Mutex
 	closed                chan struct{}
+
+	tracer trace.Tracer
```

Comment:
a.config.tracer is an Elastic APM tracer but a.tracer is an otel tracer, might be confusing to one. Is it possible to make it clearer?

Reply:
ah, I see that it was discussed. So maybe add a comment that this should not be confused with a.config.Tracer?

Reply:
Thanks for the suggestion, I approved it. Once the migration to otel is completed I think this will look way less confusing, but meanwhile a clarification comment helps for sure.